include/deal.II/non_matching/quadrature_generator.h

```diff
@@ -252,13 +259,31 @@ namespace NonMatching
     void
     set_1D_quadrature(const unsigned int q_index);
 
+  protected:
+    /**
+     * Clears the quadratures.
+     */
+    void
+    clear_quadratures();
+
+    /**
+     * Same as above but does not clear quadratures and appends it to the
+     * existing quadrature instead.
+     */
+    void
+    generate_append(const Function<dim> &   level_set,
+                    const BoundingBox<dim> &box);
+
   private:
     /**
      * QuadratureGenerator is mainly used to start up the recursive
      * algorithm. This is the object that actually generates the quadratures.
      */
     internal::QuadratureGeneratorImplementation::QGenerator<dim, dim>
       q_generator;
+
+  public:
+    friend class FaceQuadratureGenerator<dim + 1>;
```

Comment:
This would be necessary to give `FaceQuadratureGenerator` access to protected functions of `QuadratureGenerator`. I am not really a friend of `friend` declarations. In my opinion, we should make `clear_quadratures()` and `generate_append()` public here. WDYT @simonsticko ?

Reply:
Dang. I didn't think about that. I only thought about the non-face version... Okey, I also think you should go with your initial suggestion then.

Reply:
Public it is 👍🏽